{
  "_url": "https://github.com/PowerShell/PowerShell/issues/3538",
  "author": "Glober777",
  "body": "Hi!\r\n\r\nI'm having a strange issue with one of my scripts, where the New-CimSession cmdlet throws out a strange error. Basically, the line that errors out looks as follows:\r\n```Posh\r\n$CimSession = New-CimSession -ComputerName $Target\r\n\r\n# Value inside $Target defined above and confirmed to be correct\r\n```\r\nHere's the content of the generated Error Record:\r\n\r\n```\r\nException                     : Microsoft.Management.Infrastructure.CimException: An error -2147024882 was encountered\r\n                                while subscribing to a Group Policy change notification.\r\n                                   at Microsoft.Management.Infrastructure.CimException.ThrowIfMiResultFailure(MiResult result, String errorMessage, InstanceHandle errorData)\r\n                                   at Microsoft.Management.Infrastructure.CimSession.Create(String computerName, CimSessionOptions sessionOptions)\r\n                                   at Microsoft.Management.Infrastructure.CimCmdlets.CimSessionProxy.CreateSetSession(String computerName, CimSession cimSession, CimSessionOptions sessionOptions, CimOperationOptions operOptions, Boolean temporaryCimSession)\r\n                                   at Microsoft.Management.Infrastructure.CimCmdlets.CimSessionProxy..ctor(String computerName, CimSessionOptions sessionOptions)\r\n                                   at Microsoft.Management.Infrastructure.CimCmdlets.CimNewSession.NewCimSession(NewCim SessionCommand cmdlet, CimSessionOptions sessionOptions, CimCredential credential)\r\n                                   at Microsoft.Management.Infrastructure.CimCmdlets.NewCimSessionCommand.ProcessRecord()\r\n                                   at System.Management.Automation.CommandProcessor.ProcessRecord()\r\nTargetObject                  :\r\nFullyQualifiedErrorId         : Microsoft.Management.Infrastructure.CimException,Microsoft.Management.Infrastructure.CimCmdlets.NewCimSessionCommand\r\nInvocationInfo                : System.Management.Automation.InvocationInfo\r\nErrorCategory_Category        : 0\r\nErrorCategory_Activity        : New-CimSession\r\nErrorCategory_Reason          : CimException\r\nErrorCategory_TargetName      :\r\nErrorCategory_TargetType      :\r\nErrorCategory_Message         : NotSpecified: (:) [New-CimSession], CimException\r\nSerializeExtendedInfo         : False\r\nErrorDetails_ScriptStackTrace : <-Removed->\r\nPSMessageDetails              :\r\n```\r\n\r\nHere's what's inside the Exception property of the abovementioned Error Record object:\r\n\r\n```\r\nNativeErrorCode : Failed\r\nErrorData       : MSFT_WmiError\r\nMessageId       : HRESULT 0x803381d1\r\nErrorSource     :\r\nErrorType       : 0\r\nStatusCode      : 1\r\nMessage         : An error -2147024882 was encountered while subscribing to a Group Policy change notification.\r\nData            : {}\r\nInnerException  :\r\nTargetSite      : Void ThrowIfMiResultFailure(Microsoft.Management.Infrastructure.Native.MiResult, System.String, Microsoft.Management.Infrastructure.Native.InstanceHandle)\r\nStackTrace      :    at Microsoft.Management.Infrastructure.CimException.ThrowIfMiResultFailure(MiResult result, String errorMessage, InstanceHandle errorData)\r\n                     at Microsoft.Management.Infrastructure.CimSession.Create(String computerName, CimSessionOptions sessionOptions)\r\n                     at Microsoft.Management.Infrastructure.CimCmdlets.CimSessionProxy.CreateSetSession(String computerName, CimSession cimSession, CimSessionOptions sessionOptions, CimOperationOptions operOptions, Boolean temporaryCimSession)\r\n                     at Microsoft.Management.Infrastructure.CimCmdlets.CimSessionProxy..ctor(String computerName, CimSessionOptions sessionOptions)\r\n                     at Microsoft.Management.Infrastructure.CimCmdlets.CimNewSession.NewCimSession(NewCimSessionCommand cmdlet, CimSessionOptions sessionOptions, CimCredential credential)\r\n                     at Microsoft.Management.Infrastructure.CimCmdlets.NewCimSessionCommand.ProcessRecord()\r\n                     at System.Management.Automation.CommandProcessor.ProcessRecord()\r\nHelpLink        :\r\nSource          : Microsoft.Management.Infrastructure\r\nHResult         : -2146233088\r\n```\r\n\r\nThere are no inner exceptions.\r\n\r\nThe error message seems to be misleading, as the script has nothing to do with Group Policies (unless this is something that was caused by some group policy setting). \r\n\r\nThe issue occurs only if I run the script as a Scheduled Task and not if I run it interactively (running as the same RunAs account that's used to schedule it).\r\n\r\nThe script creates a new PSJob, inside which it is creating a new CimSession to a remote machine (which is available on the network) before running some queries. Unfortunately, I can't publish the whole script here.\r\n\r\nThe issue is observed on Windows Server 2008 R2 machine. PowerShell version is:\r\n```\r\nName                           Value\r\n----                           -----\r\nPSVersion                      5.0.10586.117\r\nPSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}\r\nBuildVersion                   10.0.10586.117\r\nCLRVersion                     4.0.30319.36373\r\nWSManStackVersion              3.0\r\nPSRemotingProtocolVersion      2.3\r\nSerializationVersion           1.1.0.1\r\n```\r\nWe've tried disabling the antivirus software but that made no difference here.\r\n\r\nAre there any suggestions on what may be causing this error?\r\n\r\nThanks!\r\n",
  "closed_at": "2020-02-03T00:00:30Z",
  "comments": [
    {
      "author": "MaximoTrinidad",
      "author_association": "NONE",
      "body": "@Glober777 \r\n\r\nAlthough this is not truly PowerShell (v6-Alpha) Open Source issue, I may suggest to also post this incident at Windows PowerShell UserVoice site at: https://windowsserver.uservoice.com/forums/301869-powershell/category/148044-powershell-engine \r\n\r\nThis way the Windows PowerShell team can assist in this issue.  \r\n\r\nNow, because this is part of the PowerShell Core modules, I'm sure they are going to take a look at it too.\r\n\r\nThanks!\r\n\r\n",
      "created_at": "2017-04-12T14:24:40Z",
      "updated_at": "2017-04-12T14:24:40Z"
    },
    {
      "author": "iSazonov",
      "author_association": "COLLABORATOR",
      "body": "@Glober777 Can you test in latest Windows 10 versions and maybe with latest PowerShell Core?",
      "created_at": "2017-04-12T15:10:19Z",
      "updated_at": "2017-04-12T15:10:19Z"
    },
    {
      "author": "Glober777",
      "author_association": "NONE",
      "body": "@MaximoTrinidad, thanks for the prompt reply. I've submitted the incident on UserVoice:\r\nhttps://windowsserver.uservoice.com/forums/301869-powershell/suggestions/18917938-new-cimsession-from-within-a-psjob-created-by-a-sc\r\n ",
      "created_at": "2017-04-12T15:19:56Z",
      "updated_at": "2017-04-12T15:19:56Z"
    },
    {
      "author": "Glober777",
      "author_association": "NONE",
      "body": "@iSazonov, I'll give it a try from a Windows Server 2016 box tomorrow and will tell whether it's going to be any different. I'm not 100% sure about trying through PowerShell Core, as some of the stuff that my script depends on may not be ready for that.",
      "created_at": "2017-04-12T15:22:31Z",
      "updated_at": "2017-04-12T15:22:31Z"
    },
    {
      "author": "MaximoTrinidad",
      "author_association": "NONE",
      "body": "Awesome @Glober777 and @iSazonov !  It will help for sure.\r\n:)",
      "created_at": "2017-04-12T15:49:37Z",
      "updated_at": "2017-04-12T15:49:37Z"
    },
    {
      "author": "SteveL-MSFT",
      "author_association": "MEMBER",
      "body": "This looks like an issue in the WMIv2 client layer and not in PowerShell itself.  I would suggest trying to pass an explicit credential to `new-cimsession` instead of using the current security context.",
      "created_at": "2017-04-12T15:55:47Z",
      "updated_at": "2017-04-12T15:55:47Z"
    },
    {
      "author": "Glober777",
      "author_association": "NONE",
      "body": "@SteveL-MSFT, specifying credentials explicitly didn't help. It fails with the same misleading error message.\r\n\r\n@iSazonov, running in on a Windows Server 2016 machine as a scheduled task worked like a charm (even without explicit creds). Here's the PS details from that machine:\r\n```\r\nName                           Value\r\n----                           -----\r\nPSVersion                      5.1.14393.953\r\nPSEdition                      Desktop\r\nPSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}\r\nBuildVersion                   10.0.14393.953\r\nCLRVersion                     4.0.30319.42000\r\nWSManStackVersion              3.0\r\nPSRemotingProtocolVersion      2.3\r\nSerializationVersion           1.1.0.1\r\n```\r\n\r\nAre there any other suggestions rather than moving the script over to 2016 server? :)\r\n  ",
      "created_at": "2017-04-14T11:55:15Z",
      "updated_at": "2017-04-14T11:55:15Z"
    },
    {
      "author": "Glober777",
      "author_association": "NONE",
      "body": "I could probably try upgrading WMF to 5.1 on that 2008R2 machine and see if that helps.",
      "created_at": "2017-04-14T11:56:51Z",
      "updated_at": "2017-04-14T11:56:51Z"
    },
    {
      "author": "iSazonov",
      "author_association": "COLLABORATOR",
      "body": "To make the update is good.",
      "created_at": "2017-04-14T12:02:54Z",
      "updated_at": "2017-04-14T12:02:54Z"
    },
    {
      "author": "msftbot[bot]",
      "author_association": "NONE",
      "body": "This issue has been marked as answered and has not had any activity for **1 day**. It has been closed for housekeeping purposes.",
      "created_at": "2020-02-03T00:00:29Z",
      "updated_at": "2020-02-03T00:00:29Z"
    }
  ],
  "created_at": "2017-04-12T07:37:08Z",
  "labels": [
    "WG-Cmdlets",
    "Issue-Discussion",
    "Resolution-Answered"
  ],
  "number": 3538,
  "state": "closed",
  "title": "New-CimSession from within a PSJob created by a scheduled task",
  "updated_at": "2020-02-03T00:00:30Z"
}